### DISCUSSION TITLE
Help for fix my camera brick after update model Yoosee YYZ3 type

### DISCUSSION THREAD

**Discussion by xabiruprasetyo** [e27-camera-hack/E27-Camera-Hack#7] [Q&A]
Help for fix my camera brick after update model Yoosee YYZ3 type after update from yoosee app my camera offline and not connected. no sound, lan connected after reset

try update from sdcard not working
![WhatsApp Image 2024-08-18 at 17 33 52](https://github.com/user-attachments/assets/870036f2-2a7c-42ad-8951-07ee3fa5d5eb)

I need firmware full dump for flashing from CH341A

**Comment by xabiruprasetyo** [comment]
@filder35 please help

**Reply by filder35** [reply]
I don't have a dump for your camera. But you can try to restore it. You need to know exactly what sensor and wifi. Here on the forum there are dumps for such cameras with a 300 processor. If possible, remove the boot log. To determine what hardware is used.

**Reply by xabiruprasetyo** [reply]
how can i clear the boot log
please tell me

**Comment by filder35** [comment]
The boot log is removed by connecting via UART (you can read a lot of information on the Internet). CH341A can be used for connection (you also need to look for information on the Internet) for example:
https://antenna-dvb-t2.ru/uart_windows.php
On the camera itself, you need to find the pins for connection. (if possible, a photo from both sides) In your photo, my guess is this:
![Снимок экрана в 2024-08-18 20-11-32](https://github.com/user-attachments/assets/05e062dc-a5c5-42f9-bb78-86a609ed998f)
But these are my assumptions that need to be verified.
On my camera I connected like this:
![237278014-1bece6e6-9fbd-4d39-9595-3cfb406d0cf2](https://github.com/user-attachments/assets/c4e90674-0c02-4c7c-9c66-c9fc7744d829)
What operating system are you using?

**Reply by xabiruprasetyo** [reply]
for both sides kamera 
![WhatsApp Image 2024-08-18 at 23 24 08](https://github.com/user-attachments/assets/b25787f4-d5f0-4dcb-baf7-dde88ad9d1ca)

I tried to connect my camera with ttl CH341 connect but there is no root access running in putty
![ch341-ttl-mode](https://github.com/user-attachments/assets/7d27011d-53c6-47af-a8da-1b4c19a36a4e)
![WhatsApp Image 2024-08-18 at 23 24 40](https://github.com/user-attachments/assets/e673a19c-a7a9-4a9e-b501-722014298445)

my operation system is windows 10

**Reply by filder35** [reply]
I see that there is silence in the terminal.
Did you convert CH341 to UART? It seems that some jumper needs to be changed to make it work as UART, I can't say for sure, I haven't used such a converter.

And the pins on the board are not always the same. (The location of GND Tx Rx may differ from mine) If you have a multimeter, find the ground GND. And the other two Tx Rx can be tried by enumeration, but the terminal should display. Open the terminal as administrator.
I had a CH341 board, a little later I will also try to connect and see how it all happens.

**Reply by filder35** [reply]
And check the soldering points to make sure there is no connection between the contacts.

**Reply by filder35** [reply]
And one more piece of advice, act more carefully (it is better to read a lot and do it once) the boards are very fragile, you can kill the board. Do not rush. I did everything on Linux, but you can do it on Windows too.

**Reply by filder35** [reply]
I tried my CH341
Translated it to TTL
![IMG20240819205025](https://github.com/user-attachments/assets/d858c46a-c118-4114-bec6-9537460d715e)
moved the chip to 2-3
![IMG20240819205033](https://github.com/user-attachments/assets/f804b11c-8560-43bb-88de-5026a49a47fc)
Connected GND Tx Rx
![IMG20240819213046](https://github.com/user-attachments/assets/a888efd0-8d85-49f0-807e-d1d514fb4070)
Result
![IMG20240819212950](https://github.com/user-attachments/assets/8f761072-dfb3-461d-81a3-d23a305b2148)
The first time I connected Tx Rx incorrectly, there was silence in the terminal, then I swapped them and it worked.

**Reply by xabiruprasetyo** [reply]
I will try another day. because I have other work right now

**Reply by xabiruprasetyo** [reply]
maybe this is the rx and tx port
![rxtx](https://github.com/user-attachments/assets/7e8f408c-0ac0-45cd-959f-b6e9737818b3)

**Reply by filder35** [reply]
No, it's a WiFi module, it's better not to touch it.

**Reply by filder35** [reply]
Do you have a multimeter available?

**Reply by filder35** [reply]
On a camera disconnected from power supply.
You can also check with a multimeter. Turn on the continuity test.
![Снимок экрана в 2024-08-20 15-43-38](https://github.com/user-attachments/assets/4a5c40f9-0116-474e-954b-d16679606644)
The red probe to the ground.
![Снимок экрана в 2024-08-20 15-56-56](https://github.com/user-attachments/assets/b1c19dde-18f0-48e1-9dce-0b30d928c899)

And with the black probe we check the contacts that interest us, the ground will show 0. The other two contacts TX/RX will show approximately the same value from 300-500
![Снимок экрана в 2024-08-20 15-56-56](https://github.com/user-attachments/assets/6bb6ee02-f632-4cac-9d2a-daf811f479ca)

**Comment by xabiruprasetyo** [comment]
@filder35 
After I tried many times, I finally found the right way to connect the UART.

this is my setup
![WhatsApp Image 2024-08-28 at 12 00 22](https://github.com/user-attachments/assets/43871b1d-76de-4f6e-850d-2e78033b0fd3)

i also attached the logs from my UART
[anyka_log.txt](https://github.com/user-attachments/files/16776563/anyka_log.txt)

**Reply by filder35** [reply]
We've completed the first stage. I'll study the download log and then we'll move on.

**Reply by xabiruprasetyo** [reply]
I have a backup file that I got from the FM25Q64 Flash IC series that I took using the CH341 from my board module
Do I need to upload here?

**Reply by filder35** [reply]
### At first glance, there is something wrong with your Wi-Fi drivers.
==================================================================
_Wifi type: rda5995
[RDAWLAN_DRIVER] Compiled on Mar 15 2021 at 09:25:29.
[RDAWLAN_DRIVER] Ver: 1.5.0.
[RDAWLAN_ERR]:<wland_debugfs_init,384>: debugfs is not enabled on this kernel
[RDAWLAN_ERR]:<wland_debugfs_attach,349>: root folder is NULL
[RDAWLAN_ERR]:<wland_get_mac_address,95>: read nvrame fail, ret:-1
[RDAWLAN_ERR]:<wland_nvram_write,73>: [nvram_write] : failed to open!
[RDAWLAN_ERR]:<wland_bus_start,144>: set cur_etheraddr failed!
[RDAWLAN_ERR]:<wland_usb_rx_refill,703>: usb submit rx urb fail:-19
[RDAWLAN_ERR]:<wland_usb_rx_refill,703>: usb submit rx urb fail:-19
[RDAWLAN_ERR]:<wland_usb_rx_refill,703>: usb submit rx urb fail:-19
[RDAWLAN_ERR]:<wland_usb_rx_refill,703>: usb submit rx urb fail:-19
[RDAWLAN_ERR]:<wland_usb_rx_refill,703>: usb submit rx urb fail:-19
[RDAWLAN_ERR]:<wland_usb_rx_refill,703>: usb submit rx urb fail:-19
[RDAWLAN_ERR]:<wland_usb_rx_refill,703>: usb submit rx urb fail:-19
[RDAWLAN_ERR]:<wland_usb_rx_refill,703>: usb submit rx urb fail:-19
[RDAWLAN_ERR]:<wland_usb_rx_refill,703>: usb submit rx urb fail:-19
[RDAWLAN_ERR]:<wland_usb_rx_refill,703>: usb submit rx urb fail:-19
[RDAWLAN_ERR]:<wland_usb_bus_rxctl,443>: resumed on timeout
[RDAWLAN_ERR]:<wland_proto_cdc_cmplt,468>: ***response failed, ret:-110***
[RDAWLAN_ERR]:<wland_proto_cdc_data,518>: cdc_cmplt fail.
[RDAWLAN_ERR]:<wland_run_firmware,174>: WID Result Failed
[RDAWLAN_ERR]:<wland_start_chip,540>: run firmware failed!
ifconfig: SIOCSIFFLAGS: No such device
=====================================================================================
"Configuration file: /etc/Wireless/hostapd.conf
Could not read interface wlan0 flags: No such device
ioctl(SIOCGIFINDEX): No such device
rtl871xdrv driver initialization failed."
=========================================================================================## 
Yes, if you have a dump, post it.

**Reply by xabiruprasetyo** [reply]
my dump file
FM25Q64 chip Anyka AK3918EN080 V300
https://drive.google.com/file/d/1PCSacp_eW-EZIPHA2CKMtH0XWYymX-qi/view?usp=sharing

**Comment by filder35** [comment]
Option 1.
You can try to check in the terminal (is there access to the file system or is it rooted). But to do this, you need to stop the cyclical information in the terminal. Try when the main part is loaded and cyclical information goes in the log, press "Enter" and the user input should appear. Enter the user "root" If the password entry appears, then the system is under root. If "#" appears, then we can stop the cyclical information. By killing the processes that perform the cyclical. The difficulty of entering will be that information will constantly go on the screen, but at the same time, what you enter will be processed in the terminal and will work.

**Reply by xabiruprasetyo** [reply]
yes, I can login to root
then what command should I type

**Reply by filder35** [reply]
To stop the cyclic information, you need to see which process is running it.
To see which processes are running, use the command: ps
You can post what is displayed. By stopping the cyclic display of information, we can then easily collect the information we need about the system and the installed equipment.
And the dump that you have, too, post it, I'll analyze it, we'll see what's inside, what drivers are available.

**Reply by filder35** [reply]
Ok, I downloaded the dump, I’ll analyze it and post it so that you can also navigate it.

**Reply by xabiruprasetyo** [reply]
@filder35 I will try to run the command and send the results immediately

Thank you for the help

**Reply by filder35** [reply]
Ok, don't rush, everything needs to be done carefully

**Reply by xabiruprasetyo** [reply]
this processes are running log
[processes are running.txt](https://github.com/user-attachments/files/16798372/processes.are.running.txt)

**Comment by filder35** [comment]
Everything is fine.
Now let's try to kill the main process to stop the cyclic information.
Run: killall ipc

If everything works out, then you can safely execute commands inside the system.
First, let's check which drivers are loaded with the command:
lsmod
Next, let's see how the system is mounted with two commands:
cat /proc/mtd
mount

**Reply by xabiruprasetyo** [reply]
can't run killall ipc there is an error message "killall: ipc: no process killed" and always reboots into U-Boot

```
LOG :

[root@anyka ~]$ killall ipc
killall: ipc: no process killed
[root@anyka ~]$ killall ipc

U-Boot 2013.10.0-AK_V3.0.08 (Jan 07 2020 - 21:24:50)

DRAM:  64 MiB
8 MiB
Create flash partition table init OK!
ANYKA SDHC/MMC4.0: 0
Load Env CRC OK!
In:    serial
Out:   serial
Err:   serial
no hold the upgrade button down, button value is 1
Net:   AKEthernet-0

Hit any key to stop autoboot:  1                                              0

anyka$

```

result of executing the command
"cat /proc/mtd"
"mount"

```
[root@anyka ~]$ cat /proc/mtd
dev:    size   erasesize  name
mtd0: 00800000 00001000 "spi0.0"
mtd1: 00180000 00001000 "KERNEL"
mtd2: 00001000 00001000 "MAC"
mtd3: 00001000 00001000 "ENV"
mtd4: 00100000 00001000 "A"
mtd5: 00100000 00001000 "B"
mtd6: 00433000 00001000 "C"


[root@anyka ~]$ mount
rootfs on / type rootfs (rw)
/dev/root on / type squashfs (ro,relatime)
devtmpfs on /dev type devtmpfs (rw,relatime,mode=0755)
proc on /proc type proc (rw,relatime)
tmpfs on /tmp type tmpfs (rw,relatime)
tmpfs on /var type tmpfs (rw,relatime)
devpts on /dev/pts type devpts (rw,relatime,mode=600,ptmxmode=000)
tmpfs on /mnt type tmpfs (rw,relatime)
sysfs on /sys type sysfs (rw,relatime)
/dev/mtdblock5 on /rom type jffs2 (rw,relatime)
tmpfs on /mnt/ramdisk type tmpfs (rw,relatime)
tmpfs on /etc type tmpfs (rw,relatime)
tmpfs on /tmp type tmpfs (rw,relatime)
/dev/mtdblock6 on /ipc type squashfs (ro,relatime)
/dev/mtdblock6 on /usr type squashfs (ro,relatime)
/dev/mmcblk0p1 on /mnt/disc1 type vfat (rw,relatime,fmask=0022,dmask=0022,codepage=cp437,iocharset=iso8859-1,shortname=mixed,errors=continue)
[root@anyka ~]$ mount
rootfs on / type rootfs (rw)
/dev/root on / type squashfs (ro,relatime)
devtmpfs on /dev type devtmpfs (rw,relatime,mode=0755)
proc on /proc type proc (rw,relatime)
tmpfs on /tmp type tmpfs (rw,relatime)
tmpfs on /var type tmpfs (rw,relatime)
devpts on /dev/pts type devpts (rw,relatime,mode=600,ptmxmode=000)
tmpfs on /mnt type tmpfs (rw,relatime)
sysfs on /sys type sysfs (rw,relatime)
/dev/mtdblock5 on /rom type jffs2 (rw,relatime)
tmpfs on /mnt/ramdisk type tmpfs (rw,relatime)
tmpfs on /etc type tmpfs (rw,relatime)
tmpfs on /tmp type tmpfs (rw,relatime)
/dev/mtdblock6 on /ipc type squashfs (ro,relatime)
/dev/mtdblock6 on /usr type squashfs (ro,relatime)
/dev/mmcblk0p1 on /mnt/disc1 type vfat (rw,relatime,fmask=0022,dmask=0022,codepage=cp437,iocharset=iso8859-1,shortname=mixed,errors=continue)
```

**Comment by filder35** [comment]
Got it.

Let's do another experiment.
Try this:
killall -q ipc
killall -0 shell_debug

Check which drivers are loaded with the command:
lsmod

Try to force load the wifi driver
insmod /ipc/modules/rdawfmac.ko

**Reply by filder35** [reply]
Here is an article something similar to your camera.
This is just for information. (cameras are different)
https://thao.pw/hacking-yoosee-camera?x-host=thao.pw

**Reply by xabiruprasetyo** [reply]
```
[root@anyka /ipc/modules]$ lsmod
    Tainted: G
presetmotor 6197 1 - Live 0xbf046000 (O)
otg_hs 15026 0 - Live 0xbf03e000
akgpios 3188 1 - Live 0xbf03a000
ak_info_dump 1689 0 - Live 0xbf036000
akcamera 16485 3 ak_info_dump, Live 0xbf02c000
sensor_H66 4204 0 - Live 0xbf027000
sensor_h65 3662 0 - Live 0xbf023000 (O)
sensor_H63 4160 0 - Live 0xbf01e000 (O)
sensor_h62 4198 0 - Live 0xbf019000
sensor_gc2053 4178 0 - Live 0xbf014000 (O)
sensor_f22+f23+f28+f35+f37 4187 0 - Live 0xbf00f000 (O)
sensor_c2399 4460 0 - Live 0xbf00a000
sensor_SC2239+SC2332+SC2335 5403 0 - Live 0xbf005000 (O)
sensor_SC1345 4247 0 - Live 0xbf000000 (O)


[root@anyka ~]$ insmod /ipc/modules/rdawfmac.ko
insmod: can't open '/ipc/modules/rdawfmac.ko': No such file or directory


[root@anyka /ipc/modules]$ ls
ak_info_dump.ko                 sensor_f23_f37.ko
akcamera.ko                     sensor_gc2053.ko
akgpios.ko                      sensor_h62.ko
otg-hs.ko                       sensor_h63.ko
sensor_SC1345.ko                sensor_h65.ko
sensor_SC2239+SC2332+SC2335.ko  sensor_h66.ko
sensor_c2399.ko                 wifi

[root@anyka /ipc/modules]$ 
```

**Reply by filder35** [reply]
when launching the wifi driver, I gave the wrong command
so try this:
insmod /ipc/modules/wifi/rdawfmac.ko

**Reply by xabiruprasetyo** [reply]
I tried typing "insmod /ipc/modules/wifi/rdawfmac.ko"
and the results

```
[root@anyka ~]$ insmod /ipc/modules/wifi/rdawfmac.ko
insmod: can't insert '/ipc/modules/wifi/rdawfmac.ko': File exists
```

**Reply by xabiruprasetyo** [reply]
Is it possible that my WiFi driver is corrupt?
then where can I get the appropriate wifi driver and how can I put it in system /ipc/modules/wifi/

**Reply by filder35** [reply]
Possibly.

But I need to check for sure. Give me some time, I'll try to find analog drivers for your system. I see that your SD is mounted normally. Let's try to run it via SD.
The error says that the driver is running, but it is not displayed in lsmod.

Either it's software or mechanical damage to the WiFi module.
You can try this:
modinfo /ipc/modules/wifi/rdawfmac.ko

**Comment by filder35** [comment]
Here is your firmware disassembled, you can see what's inside
[file_out.zip](https://github.com/user-attachments/files/16815895/file_out.zip)

**Reply by xabiruprasetyo** [reply]
how to extract the firmware_dump that we get from the ISP Programmer

**Comment by filder35** [comment]
https://cloud.gwell.cc/?service_27/232.html
here is the site where your camera points to for an update, you can also write there, they can send you the correct update

as an option

**Comment by filder35** [comment]
So they are
https://aliexpress.ru/item/32960451533.html?sku_id=66500492969&spm=a2g2w.productlist.search_results.0.675d11e5SwMJ5V
But I'm not sure that it physically died.
Maybe software-wise. Before the update the camera worked, after the update it stopped. It's possible that the update didn't arrive correctly.

Try contacting the link I posted, this is the link where the updates come from. It's just that somewhere on the forum an analog camera also died after the update and they provided the files for the correct update. (as an option)

**Comment by xabiruprasetyo** [comment]
Thank You @filder35 
Now my camera can be online again
many thanks for the help

![CCTV_Online](https://github.com/user-attachments/assets/e517f2e9-510d-4b82-85cc-b49e64fa435a)
![WhatsApp Image 2024-08-31 at 12 21 55](https://github.com/user-attachments/assets/d99ac132-7004-457f-8fb5-e235c7ce0782)

**Reply by filder35** [reply]
It's good that it worked out well.
Check all the launch systems again and assemble the camera if everything works.

**Reply by xabiruprasetyo** [reply]
I've checked again and made sure it is working properly and the system load is running well on the UART monitor
now the camera is installed

**Reply by xabiruprasetyo** [reply]
I have a question.
I have another camera that uses the GM8135S chip. Where can I ask about that camera?

**Comment by filder35** [comment]
You can also connect via terminal and see what the loading shows.
https://github.com/zzerrg/gmfwtools
https://www.cnx-software.com/2015/06/30/grain-media-8136s-8138s-are-low-cost-hd-ip-camera-socs/
https://zftlab.org/pages/2018010500.html   

You asked how I disassembled the firmware?

**Reply by xabiruprasetyo** [reply]
ok I check the website

yes, how to disassembled firmware?

**Reply by filder35** [reply]
ok I'll post how it's done in Linux

**Reply by xabiruprasetyo** [reply]
OK, I'll wait for the post

**Comment by filder35** [comment]
Unzipping the firmware of cameras under the ANYKA processor.

I did all this under Linux.
https://drive.google.com/file/d/1-iHICuCO92STRmJH8GWREhKjBiz2yC8U/view?usp=sharing
I use this system, I like it, it has never failed with different architectures for assembling, disassembling and compiling applications and systems.

Let's get started.

We have a dump of the firmware of the IP camera removed by the programmer or by means of ftp.

First, we will need some programs:

binwalk -- a good program with additional applications necessary for reverse engineering.
jefferson --program for unzipping (I use this program to unzip a block with
jffs2 this block usually contains configuration files and in the camera it is in -rw format editing is allowed
and I usually do not assemble it back it can be edited on the board.

binwalk - installed by commands:
#sudo apt update
#sudo apt install binwalk
(more detailed information on the program's capabilities can be found on the Internet)

jefferson - installed by commands:
#git clone https://github.com/sviehb/jefferson.git
#cd jefferson
#sudo apt update
#sudo apt install python3-pip liblzo2-dev
#sudo python3 -m pip install -r requirements.txt
#sudo python3 setup.py install
(more detailed information can be found on the Internet)

Next, we will study the dump
(I did it this way: created a separate folder for the project, put it, copied the dump there and in it performed actions)

#binwalk -t dump_DUMP.bin
![Снимок экрана в 2024-09-01 14-48-53](https://github.com/user-attachments/assets/da178aa3-c0d1-4001-95be-614579e259af)

I allocated blocks in the firmware

kernel - we don't need
rootfs - root file system
jffs2/rom - mounted system
usr/ipc - user mounted system

how it is mounted on the camera can be seen by the commands
#cat /proc/mtd
#mount

/dev/mtdblock5 on /rom type jffs2 (rw,relatime)
/dev/mtdblock6 on /ipc type squashfs (ro,relatime)
/dev/mtdblock6 on /usr type squashfs (ro,relatime)

We will analyze it with the command dd
Let's extract the rootfs block

#dd if=DUMP.bin of=root skip=1810432 bs=1 count=931284
(where skip=1810432 is the offset in the figure, underlined, count=931284 is the size in the figure, this is size: 931284 bytes)
a root file will appear in the folder
If you check the root file with the binwalk utility, you will see that the file is archived
![Снимок экрана в 2024-09-01 16-43-52](https://github.com/user-attachments/assets/a4bd592c-068c-475a-9500-12e4a4d7f77e)

unzip it with the utility included in binwalk
#unsquashfs -d root_out root
A root_out folder will appear, where all the files are actually located in an accessible form. (what's interesting there are the configuration files in the /etc folder, for me it's more interesting the files in the /etc/init.d folder
rcS, rc.local in them you can see the order of system startup 1 file is rcS, then rc.local. In them you can see the order of system startup, mounting and subsequent startup of configuration files and where they are located in the system.)

Pull out the usr/ipc block

#dd if=DUMP.bin of=usr skip=3907584 bs=1 count=4344916
(the usr file will appear, here everything is the same as with the root file.)
#unsquashfs -d usr_out usr
(the usr_out folder will appear in it drivers, there are also executable files and configuration files)

Pull out the jffs2 block

#dd if=DUMP.bin of=jffs2 skip=2859008 bs=1 count=1048576
(here I skip=2859008 took from the beginning of the offset and count=1048576 calculated from the end of the block offset 3907584-2859008=1048576 got the size)

Unzip the resulting jffs2 file with the jefferson program.

#jefferson -d jffs2_out jffs2

The jffs2_out folder contains configuration files.

P.S If something is not clear, ask, the translator may not convey the information correctly.

**Reply by xabiruprasetyo** [reply]
@filder35 Thank You above post how to disassembled firmware

I will try installing Linux first and try to disassembled the firmware

**Reply by filder35** [reply]
If you install Linux which I posted above, the application advises to update and if it asks to update the system version, do not update. But the system is normal working, at least I have not had any problems with it.

**Reply by xabiruprasetyo** [reply]
Thank you for the advice about updating the system. I will not update the system
